app/src/main/java/org/onionshare/android/server/WebserverManager.kt

```diff
@@ -124,6 +124,8 @@ class WebserverManager @Inject constructor() {
                 Attachment.withParameter(FileName, sendPage.fileName).toString()
             )
             call.respondFile(sendPage.zipFile)
+            LOG.info("Download complete. Emitting SHOULD_STOP state...")
+            state.value = SHOULD_STOP
```

Comment:
I am not a 100% sure, but in my tests, also with larger files, the webserver was only shutting down after the file was transferred completely. It might not be classical blocking, but suspending of the function until the call has responded, so that the next call is only made afterwards not simultaneously.